queries.sql

```diff
@@ -22,3 +22,93 @@ WHERE name <> 'Gabumon';
 
 SELECT * FROM animals
 WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- First transaction
+
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+-- Second transaction
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+-- Third transaction
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+-- forth transaction
+
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT save_point_1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO save_point_1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+COMMIT;
+
+-- Answers
+
+-- How many animals are there?
+SELECT COUNT(*) AS total_animals
+FROM animals;
+
+--How many animals have never tried to escape?
+SELECT COUNT(*) AS animals_never_tried_to_escape
+FROM animals
+WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) AS average_weight
+FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT neutered, SUM(escape_attempts) AS total_escape_attempts
```

Comment:
- [ OPTIONAL ] 👋 Hey! To address the question about which animals, neutered or not, have escaped the most, it might be more informative to return the specific animal with the highest number of escape attempts 📈, rather than just the sum of escape attempts. 🐾 This way, we can pinpoint the individual animal that has had the most escapes, offering a more precise and insightful answer to the query. 🐕🦴🐈🏃‍♂️ Thank you for considering this improvement! 🙌😊
- 
```suggestion
 SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```

Reply:
Done, thanks for you review 👍😃